core/headerchain.go

```diff
@@ -92,24 +92,23 @@ func NewHeaderChain(db ethdb.Database, engine consensus.Engine, chainConfig *par
 	return hc, nil
 }
 
-func (hc *HeaderChain) ManifestForHeader(h *types.Header) (types.BlockManifest, error) {
+func (hc *HeaderChain) CollectBlockManifest(h *types.Header) (types.BlockManifest, error) {
 	manifest := types.BlockManifest{}
-	if h.NumberU64() == 0 {
+	// Terminate the search on coincidence or genesis
+	if hc.engine.HasCoincidentDifficulty(h) || h.NumberU64() == 0 {
```

Comment:
This would probably be better to use a hash check rather than a number check.

Reply:
Fixed: [5ec5fe9](https://github.com/dominant-strategies/go-quai/pull/463/commits/5ec5fe9f33328f38b04c285c27c24c65bb02a90c)